src/sqlfluff/core/linter.py

```diff
@@ -923,99 +923,102 @@ def extract_ignore_from_comment(comment: RawSegment):
                 return (comment.pos_marker.line_no, None)
         return None
 
-    def lint(
-        self, parsed: BaseSegment, config: Optional[FluffConfig] = None
-    ) -> List[SQLLintError]:
-        """Lint a parsed file object."""
-        config = config or self.config
-        linting_errors = []
-        for crawler in self.get_ruleset(config=config):
-            lerrs, _, _, _ = crawler.crawl(parsed, dialect=config.get("dialect_obj"))
-            linting_errors += lerrs
-        return linting_errors
+    @staticmethod
+    def _warn_unfixable(code: str):
+        linter_logger.warning(
+            f"One fix for {code} not applied, it would re-cause the same error."
+        )
 
-    def fix(self, parsed: BaseSegment, config: Optional[FluffConfig] = None):
-        """Fix a parsed file object."""
-        # Set up our config
+    def lint_fix(
```

Comment:
I merged lint and fix functions as they had a fair bit of repetition.

Reply:
Nice work on this refactor - much tidier